### PR TITLE
Run evm2wasm on EVM input

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

**Comment by axic** [comment]
@chfast can you add a cmake rule to compile `evm2wasm` and include its root for the header? Note: `evm2wasm` doesn't have a header, yet - I'll fix that later.

**Comment by chfast** [comment]
It needs most of the changes to be done in evm2wasm. Let me know when they are ready to check the CMake there.

**Comment by axic** [comment]
@chfast evm2wasm stuff is merged

File: src/hera.cpp [hunk 1]
```diff
@@ -52,6 +54,12 @@ void execute(
 ) {
   cout << "Executing...\n";
 
+  // Translate EVM bytcode to WASM
+  if (code[0] != 0 || code[1] != 'a' || code[2] != 's' || code[3] != 'm') {
+    std::string str_code = evm2wasm(std::string{code.data(), code.size()});
```
Comment (axic) [inline src/hera.cpp]: Has `using std`.

**Comment by axic** [comment]
Can you add `-DEVM2WASM=1` option to cmake?

File: src/hera.cpp (hunk 1 above)
Comment (axic) [reply src/hera.cpp]: Would name it actually `translated`.

File: src/hera.cpp [hunk 2]
```diff
@@ -52,6 +54,12 @@ void execute(
 ) {
   cout << "Executing...\n";
 
+  // Translate EVM bytcode to WASM
+  if (code[0] != 0 || code[1] != 'a' || code[2] != 's' || code[3] != 'm') {
```
Comment (axic) [inline src/hera.cpp]: Need to check for `code.size() >= 4`.

File: src/hera.cpp [hunk 3]
```diff
@@ -52,6 +56,14 @@ void execute(
 ) {
   cout << "Executing...\n";
 
+#if HERA_EVM2WASM
+  // Translate EVM bytecode to WASM
+  if ((code.size() >= 4) && (code[0] != 0 || code[1] != 'a' || code[2] != 's' || code[3] != 'm')) {
```
Comment (axic) [inline src/hera.cpp]: Ah this should be if `code.size() < 4 || ...`

**Comment by axic** [comment]
@chfast do you want to review this? It is not important, won't be merged soon.

**Pull request merged**